lib/rules/no-useless-escape.js

```diff
@@ -23,22 +36,9 @@ const VALID_STRING_ESCAPES = [
     "\r"
 ];
 
-const VALID_REGEX_ESCAPES = [
+const REGEX_GENERAL_ESCAPES = new Set([
     "\\",
-    ".",
-    "-",
     "^",
-    "$",
-    "*",
-    "+",
-    "?",
-    "{",
-    "}",
-    "[",
-    "]",
-    "|",
-    "(",
-    ")",
     "b",
     "B",
```

Comment:
The escape of `B` in character class is unnecessary.

```js
/[\B]/; // this escape is unnecessary; the `B` is a character.
```

On the other hand, the escape of `\b` is necessary. `\b` in character class is meaning backspace character. (`\b` outside of character class is a word boundary)